libp2p/crypto/crypto.nim

```diff
@@ -561,18 +565,47 @@ proc ephemeral*(scheme: ECDHEScheme): KeyPair =
   result.seckey.eckey = keypair.seckey
   result.pubkey.eckey = keypair.pubkey
 
+proc ephemeral*(scheme: string): KeyPair {.inline.} =
+  ## Generate ephemeral keys used to perform ECDHE using string encoding.
+  ##
+  ## Currently supported encoding strings are P-256, P-384, P-521, if encoding
+  ## string is not supported P-521 key will be generated.
+  if scheme == "P-256":
+    result = ephemeral(Secp256r1)
```

Comment:
same here - no need to use `result` - you can just remove all `result =` with the advantage that the compiler will give an error message if you have a dangling `else` or some other mis-indentation issue

Reply:
@cheatfate this is mostly your code any opinion on this? Otherwise I'll start changing this according to @arnetheduck suggestions.

Reply:
worth keeping in mind that some of these things were also not working with earlier compiler versions, but things have been improving

Reply:
My opinion is classic style with **explicit** `result = X` is much more visible and understandable for people who are not familiar with Nim, and so it will be much easy to read and understand without any visits to Nim Manual. Also because **explicit** result is working everywhere (even inside `{.async.}` transformations) it keep coding style equal in all procedures.

Reply:
Honestly, I leaning a bit more towards the `result` style. I think @cheatfate  has a point in that this is more familiar style across the board. IMO, leaving `result` out in simple constructs makes sense, but it does provide more visibility in more involved code blocks, besides that I've found it to be a handy construct, no need to explicitly create return objects, etc... If anything I would suggest we work on well defined style guidelines that we then stick too across the entire codebase, but I'd leave that for a separate effort altogether. 

Thoughts?

Reply:
well, like mentioned, you're losing out on help that the compiler can give you, as far as objective reasons go - but this is predominantly a taste and habit matter (what code you're used to seeing). at some point we'll encode what we have in common in a guide indeed and go from there. now you know both styles exist - why have a single way of doing the same thing when you can have 3 :)

Reply:
heh, TIMTOWDI FFW.

Reply:
> not familiar with Nim

@cheatfate this depends where they're coming from - people not familiar with Nim will not understand what `result =` is: Nim is unique in having this syntax for returning things, of the languages that are in use today (yes, pascal has it, and maybe some other weird ones). it was added to nim in part because it's an easy way to implement return-value-optimization, but it's no longer needed for this purpose.

implicit return is common in many languages (haskell, perl, ruby, rust etc), as is explicit `return x` (everything else)

Reply:
One reasonable middle-ground that gets you the benefits of Jacek's style and still looks more ordinary is to do it like this:

```nim
result = if foo: bar
         else: baz
```

... or the equivalent with return.